pkg/keymanagementprovider/azurekeyvault/provider.go

```diff
@@ -233,33 +278,66 @@ func (s *akvKMProvider) GetKeys(ctx context.Context) (map[keymanagementprovider.
 
 		// fetch the key object from Key Vault
 		startTime := time.Now()
-		keyResponse, err := s.keyKVClient.GetKey(ctx, keyVaultKey.Name, keyVaultKey.Version)
-		if err != nil {
-			return nil, nil, fmt.Errorf("failed to get key objectName:%s, objectVersion:%s, error: %w", keyVaultKey.Name, keyVaultKey.Version, err)
+		// if versionHistoryLimit is not set, set it to default value 1
+		if keyVaultKey.VersionHistoryLimit == 0 {
+			keyVaultKey.VersionHistoryLimit = versionHistoryLimitDefault
 		}
-		keyBundle := keyResponse.KeyBundle
-		isEnabled := *keyBundle.Attributes.Enabled
-		// if version is set as "" in the config, use the version from the key bundle
-		keyVaultKey.Version = getObjectVersion(string(*keyBundle.Key.KID))
-
-		if !isEnabled {
-			startTime := time.Now()
-			lastRefreshed := startTime.Format(time.RFC3339)
-			properties := getStatusProperty(keyVaultKey.Name, keyVaultKey.Version, lastRefreshed, isEnabled)
-			keysStatus = append(keysStatus, properties)
-			mapKey := keymanagementprovider.KMPMapKey{Name: keyVaultKey.Name, Version: keyVaultKey.Version, Enabled: isEnabled}
-			keymanagementprovider.DeleteKeyFromMap(s.resource, mapKey)
-			continue
+
+		var versionHistory []struct {
+			Version string
+			Created time.Time
 		}
 
-		publicKey, err := getKeyFromKeyBundle(keyBundle)
-		if err != nil {
-			return nil, nil, fmt.Errorf("failed to get key from key bundle:%w", err)
+		keyVersionPager := s.keyKVClient.NewListKeyVersionsPager(keyVaultKey.Name, nil)
+		for keyVersionPager.More() {
+			pager, err := keyVersionPager.NextPage(ctx)
+			if err != nil {
+				return nil, nil, fmt.Errorf("failed to get key versions for objectName:%s, error: %w", keyVaultKey.Name, err)
+			}
+			for _, key := range pager.Value {
+				versionHistory = append(versionHistory, struct {
+					Version string
+					Created time.Time
+				}{Version: key.KID.Version(), Created: *key.Attributes.Created})
+			}
+		}
+
+		// sort the version history by created time
+		sortVersionHistory(versionHistory)
+		sortedVersionHistory := GetSortedVersions(versionHistory)
+
+		// if versionHistoryLimit is greater than the number of versions, set it to the number of versions
+		if keyVaultKey.VersionHistoryLimit > len(sortedVersionHistory) {
```

Comment:
After looking at the logic again, if the length of the sortedVersions is zero then it should log that no versions were found and continue instead of using the defaultValue of 1 and moving on. Here's the logic I added to move it into a single place:

```go
		// sort the version history by created time
		sortVersionHistory(versionHistory)
		sortedVersionHistory := GetSortedVersions(versionHistory)

		if len(sortedVersionHistory) == 0 {
			logger.GetLogger(ctx, logOpt).Infof("no versions found for certificate %s", keyVaultCert.Name)
			continue
		}

		// Ensure that the versionHistoryLimit is not greater than the number of versions to avoid out of bounds error
		if keyVaultCert.VersionHistoryLimit > len(sortedVersionHistory) {
			keyVaultCert.VersionHistoryLimit = len(sortedVersionHistory)
		}
```